Missing "`impl Trait` introduces an implicit type parameter" when the trait comes from another crate

When a trait has a method that has type parameters (`fn foo<T>()`) and implicit type parameters (`fn foo(x: impl Clone)`), if you impl the trait, the compiler gives the error:

```
error[E0049]: method `foo` has 2 type parameters but its trait declaration has 3 type parameters
  --> crate1/src/main.rs:10:12
   |
4  |     fn foo<A, B>(x: u32, y: impl Debug);
   |            -  -             ----------
   |            |                |
   |            |                `impl Trait` introduces an implicit type parameter
   |            expected 3 type parameters
...
10 |     fn foo<A, B>(x: u32, y: u32) {
   |            ^  ^
   |            |
   |            found 2 type parameters
```

This is fine but if the trait comes from another crate, the compiler will just say:

```
error[E0049]: method `foo` has 2 type parameters but its trait declaration has 3 type parameters
 --> crate1/src/main.rs:6:12
  |
6 |     fn foo<A, B>(x: u32, y: u32) {
  |            ^  ^
  |            |
  |            found 2 type parameters, expected 3
```

Here is a repository with this sample code: https://github.com/cecton/rust-type-param-error

I expected to see this happen: it would be great to see that there are implicit type parameters even if it comes from another crate or at least a warning.

Instead, this happened: I see that there should be 3 type parameters and I provided 2 but if I look at the code I see 2 type parameters, not 3. It was not obvious to me that the impl in parameter was introducing a type parameter.

### Meta

`rustc --version --verbose`:
```
rustc 1.43.0-nightly (158127853 2020-03-10)
binary: rustc
commit-hash: 15812785344d913d779d9738fe3cca8de56f71d5
commit-date: 2020-03-10
host: x86_64-unknown-linux-gnu
release: 1.43.0-nightly
LLVM version: 9.0
```

--- a/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs
+++ b/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs
@@ -1557,38 +1557,24 @@ fn compare_number_of_generics<'tcx>(
                 DiagnosticId::Error("E0049".into()),
             );
 
-            let mut suffix = None;
-
+            let msg =
+                format!("expected {trait_count} {kind} parameter{}", pluralize!(trait_count),);
             if let Some(spans) = trait_spans {
                 let mut spans = spans.iter();
                 if let Some(span) = spans.next() {
-                    err.span_label(
-                        *span,
-                        format!(
-                            "expected {} {} parameter{}",
-                            trait_count,
-                            kind,
-                            pluralize!(trait_count),
-                        ),
-                    );
+                    err.span_label(*span, msg);
                 }
                 for span in spans {
                     err.span_label(*span, "");
                 }
             } else {
-                suffix = Some(format!(", expected {trait_count}"));
+                err.span_label(tcx.def_span(trait_.def_id), msg);
             }
 
             if let Some(span) = span {
                 err.span_label(
                     span,
-                    format!(
-                        "found {} {} parameter{}{}",
-                        impl_count,
-                        kind,
-                        pluralize!(impl_count),
-                        suffix.unwrap_or_default(),
-                    ),
+                    format!("found {} {} parameter{}", impl_count, kind, pluralize!(impl_count),),
                 );
             }
 

--- a/tests/ui/typeck/issue-36708.stderr
+++ b/tests/ui/typeck/issue-36708.stderr
@@ -2,7 +2,12 @@ error[E0049]: method `foo` has 1 type parameter but its trait declaration has 0 
   --> $DIR/issue-36708.rs:8:12
    |
 LL |     fn foo<T>() {}
-   |            ^ found 1 type parameter, expected 0
+   |            ^ found 1 type parameter
+   |
+  ::: $DIR/auxiliary/issue-36708.rs:4:5
+   |
+LL |     fn foo();
+   |     --------- expected 0 type parameters
 
 error: aborting due to previous error
 

